tools/check_release_notes.sh

```diff
@@ -0,0 +1,178 @@
+        print_error_and_exit "PULL_NUMBER is a required option. It must match the GitHub pull request number."
+        exit 1
+    fi
+}
+
+# Curl the GitHub API to get a list of files for the specified PR. If files are
+# found, exit. We might eventually want to validate the data here.
+checkForFiles() {
+    echo "Requesting files from pull request ${REPO_OWNER}/${REPO_NAME}#${PULL_NUMBER}"
+    if [ -z "${token}" ]; then
+        ghFiles=$(curl -L -s --show-error --fail \
+            --header "Accept: application/vnd.github.v3+json" \
+            "https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/pulls/${PULL_NUMBER}/files")
+    else
+        ghFiles=$(curl -L -s --show-error --fail \
+            --header "Accept: application/vnd.github.v3+json" \
+            --header "Authorization: Bearer ${token}" \
+            "https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/pulls/${PULL_NUMBER}/files")
+    fi
+
+    #if return code isn't 0, print the error and exit.
+    #shellcheck disable=SC2181
+    if [ $? != 0 ]; then
+        exit 1
+    fi
+
+    releaseNotesFiles=$(echo "${ghFiles}" | jq 'map(.filename)' |
+            grep 'releasenotes\/notes\/.*\.yaml' )
+
+    if [ -z "${releaseNotesFiles}" ]; then
+        echo "No release notes files found."
+    else
+        echo "Found release notes entries"
+        exit 0
+    fi
+}
+
+checkForLabel() {
+    # If no release notes files were found in the PR, go back and request the actual
+    # PR info (instead of the files) and check the labels.
+    echo "Requesting labels from pull request ${REPO_OWNER}/${REPO_NAME}#${PULL_NUMBER}"
+
+    if [ -z "${token}" ]; then
+        ghPR=$(curl -L -s --show-error --fail  \
+            -H "Accept: application/vnd.github.v3+json" \
+            "https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/pulls/${PULL_NUMBER}")
+
+    else
+        ghPR=$(curl -L -s --show-error --fail \
+            --header "Accept: application/vnd.github.v3+json" \
+            --header "Authorization: Bearer ${token}" \
+            "https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/pulls/${PULL_NUMBER}")
+
+    fi
+
+    #if return code isn't 0, print the error and exit.
+    #shellcheck disable=SC2181
+    if [ $? != 0 ]; then
+        exit 1
+    fi
+
+    labels=$(echo "${ghPR}" | jq '.labels | map(.name)')
+    releaseNotesLabelPresent=$(echo "${labels}" | grep $RELEASE_NOTES_NONE_LABEL)
+
+    if [ -z "${releaseNotesLabelPresent}" ]; then
+        echo "Missing release notes and missing release notes label. If this pull request contains user facing changes, please follow the instructions at https://github.com/istio/istio/tree/master/releasenotes to add an entry. If not, please add the release-notes-none label to the pull request"
+        exit 1
+    fi
+}
+
+main() {
+    trap cleanup EXIT
+    get_opts "$@"
+    validate_opts
+
+    checkForFiles
+    checkForLabel
+    return 1
+}
+
+main "$@"
+exit "${exit_code:-0}"
```

Comment:
Unless you intend to set `exit_code` somewhere in the code above, this line can be omitted.

Reply:
Removed. Thanks :)